### PR TITLE
TF 2.16 with Torch 2.4.0

### PR DESCRIPTION
**Pull request by calderjo**
Upgraded our base cpu and gpu base image

-updated torch eco.
-removed patch and pins placed for tf 2.15
-removed torch text, it is no longer maintained and incompatible with torch 2.4.0
-geopanda depreciated it's datasets methods, test needed to be updated

### CODE REVIEW

File: Dockerfile.tmpl [hunk 1]
```diff
@@ -122,21 +119,20 @@ RUN pip install spacy && \
 {{ end}}
 
 # Install PyTorch
+# b/356397043: magma-cuda121 is the latest version
 {{ if eq .Accelerator "gpu" }}
 COPY --from=torch_whl /tmp/whl/*.whl /tmp/torch/
-RUN mamba install -y -c pytorch magma-cuda${CUDA_MAJOR_VERSION}${CUDA_MINOR_VERSION} && \
```
Comment (djherbis) [inline Dockerfile.tmpl:127]: I think we should comment here that we're on 12.3 but that magma-cuda123 doesn't exist?

File: packages/torch.Dockerfile [hunk 2]
```diff
@@ -20,7 +19,7 @@ RUN conda install -c conda-forge mamba
 
 # Build instructions: https://github.com/pytorch/pytorch#from-source
 RUN mamba install astunparse numpy ninja pyyaml mkl mkl-include setuptools cmake cffi typing_extensions future six requests dataclasses
-RUN mamba install -c pytorch magma-cuda${CUDA_MAJOR_VERSION}${CUDA_MINOR_VERSION}
+RUN mamba install -c pytorch magma-cuda121
```
Comment (djherbis) [inline packages/torch.Dockerfile:22]: Same comment here.

**Review by djherbis — APPROVED** [review]
Awesome work! Feel free to merge and add the other comments in a follow-up, I'd rather get this ready for release :)

File: Dockerfile.tmpl (hunk 1 above)
Comment (calderjo) [reply Dockerfile.tmpl:127]: ack , in a follow up

File: packages/torch.Dockerfile (hunk 2 above)
Comment (calderjo) [reply packages/torch.Dockerfile:22]: ditto

**Pull request merged**